Sources/MetriportSDK/MetriportClient.swift

```diff
@@ -344,13 +334,97 @@ public class MetriportClient {
         return statistics.averageQuantity()
     }
 
-    private func setLocalKeyValue(key: String, val: Date) {
+    private func setLocalKeyValue(key: String, val: Any) {
         do {
-            let data : Data = try NSKeyedArchiver.archivedData(withRootObject: val as Any, requiringSecureCoding: false)
+            let data : Data = try NSKeyedArchiver.archivedData(withRootObject: val, requiringSecureCoding: false)
             UserDefaults.standard.set(data, forKey: key)
         } catch {
             print("Couldnt write files")
         }
     }
+    
+    func fetchSleep(metriportUserId: String, group: DispatchGroup) {
+        if let sleepType = HKObjectType.categoryType(forIdentifier: HKCategoryTypeIdentifier.sleepAnalysis) {
+            let calendar = Calendar.current
+            let endDate = Date()
+            let oneMonthAgo = DateComponents(day: -30)
+            guard let startDate = calendar.date(byAdding: oneMonthAgo, to: endDate) else {
+                fatalError("*** Unable to calculate the start date ***")
+            }
+            let predicate = HKQuery.predicateForSamples(withStart: startDate, end: endDate, options: [])
+            
+            var anchor = HKQueryAnchor.init(fromValue: 0)
+            
+            if UserDefaults.standard.object(forKey: "sleepAnchor") != nil {
```

Comment:
nit: let's throw `sleepAnchor` in a const, generally whenever we hardcode a string more than once, a const is a good idea